frontend/src/metabase/hooks/use-palette.ts

```diff
@@ -0,0 +1,25 @@
+import { useMemo } from "react";
+
+import type { ColorPalette } from "metabase/lib/colors/types";
+
+import { useMantineTheme, type MantineColor } from "../ui";
+
+/**
+ * Gets a color palette from the current Mantine theme.
+ *
+ * Allows palettes to be overridden by the user, primarily via the React embedding SDK.
+ */
+export function usePalette(): ColorPalette {
+  const theme = useMantineTheme();
+
+  return useMemo(() => {
+    const color = (name: MantineColor) => theme.colors[name]?.[0];
```

Comment:
Not sure if it makes sense to use the first color in the array `[0]` here as Mantine has 10 shades. The first color is the lightest color

> Colors are exposed on the theme object as an array of strings, you can access color shade by color name and index (0-9), colors with larger index are darker:

https://mantine.dev/theming/colors/